src/QMCWaveFunctions/BsplineFactory/SplineR2R.cpp

```diff
@@ -100,29 +100,17 @@ void SplineR2R<ST>::applyRotation(const ValueMatrix& rot_mat, bool use_stored_co
   if (!use_stored_copy)
     std::copy_n(spl_coefs, coefs_tot_size, coef_copy_->begin());
 
-  // Fill top left corner of tmpU with rot_mat
-  ValueMatrix tmpU;
-  tmpU.resize(Nsplines, Nsplines);
-  std::fill(tmpU.begin(), tmpU.end(), 0.0);
-  for (auto i = 0; i < rot_mat.size1(); i++)
-  {
-    for (auto j = 0; j < rot_mat.size2(); j++)
-    {
-      tmpU[i][j] = rot_mat[i][j];
-    }
-  }
-
   // Apply rotation the dumb way b/c I can't get BLAS::gemm to work...
```

Comment:
Please add a note about mismatch. What is padded and what is not.

Reply:
Also
```
assert(OrbitalSetSize == rot_mat.rows())
assert(OrbitalSetSize == rot_mat.cols())
```

Reply:
I didn't see the node as they are not close to the changed part and github review hides them.